sdk/src/types/block/payload/transaction/essence/regular.rs

```diff
@@ -419,17 +433,78 @@ fn verify_payload_packable<const VERIFY: bool>(
     Ok(())
 }
 
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+#[non_exhaustive]
+pub enum TransactionCapabilityFlag {
+    BurnNativeTokens,
+    BurnMana,
+    DestroyAccountOutputs,
+    DestroyFoundryOutputs,
+    DestroyNftOutputs,
+}
+
+impl TransactionCapabilityFlag {
+    const BURN_NATIVE_TOKENS: u8 = 0b00000001;
+    const BURN_MANA: u8 = 0b00000010;
+    const DESTROY_ACCOUNT_OUTPUTS: u8 = 0b00000100;
+    const DESTROY_FOUNDRY_OUTPUTS: u8 = 0b00001000;
+    const DESTROY_NFT_OUTPUTS: u8 = 0b00010000;
+}
```

Comment:
why not:
```rust
#[derive(Copy, Clone, Eq, PartialEq, Debug)]
#[non_exhaustive]
#[repr(u8)]
pub enum TransactionCapabilityFlag {
    BurnNativeTokens = 0b00000001,
    BurnMana = 0b00000010,
    DestroyAccountOutputs = 0b00000100,
    DestroyFoundryOutputs = 0b00001000,
    DestroyNftOutputs = 0b00010000,
}
```

```rust
impl CapabilityFlag for TransactionCapabilityFlag {
    type Iterator = core::array::IntoIter<Self, 5>;

    fn as_byte(&self) -> u8 {
        *self as u8
    }
.
.
.
```

Reply:
Because there will be duplicates in the constants

Reply:
but do we still need the constants then?

Reply:
Not sure to follow, where do you put the values if not in the constants?

Reply:
in the enum like above?

Reply:
As said, there are duplicates so enum is not an option